crates/ruma-common/src/events/content.rs

```diff
@@ -101,29 +52,38 @@ pub trait RoomAccountDataEventContent: EventContent<EventType = RoomAccountDataE
 /// Content of an ephemeral room event.
 pub trait EphemeralRoomEventContent: EventContent<EventType = EphemeralRoomEventType> {}
 
-/// Content of a message-like event.
+/// Content of a non-redacted message-like event.
 pub trait MessageLikeEventContent: EventContent<EventType = MessageLikeEventType> {}
 
 /// Content of a redacted message-like event.
-pub trait RedactedMessageLikeEventContent: MessageLikeEventContent {}
+pub trait RedactedMessageLikeEventContent: EventContent<EventType = MessageLikeEventType> {}
 
-/// Content of a state event.
+/// Content of a non-redacted state event.
 pub trait StateEventContent: EventContent<EventType = StateEventType> {
     /// The type of the event's `state_key` field.
     type StateKey: AsRef<str> + Clone + fmt::Debug + DeserializeOwned + Serialize;
 }
 
-/// Content of a non-redacted state event.
-pub trait OriginalStateEventContent: StateEventContent + RedactContent {
+/// Content of a non-redacted state event with a corresponding possibly-redacted type.
+pub trait StaticStateEventContent: StateEventContent {
```

Comment:
Is there value in having this in a separate trait than `StateEventContent`?

Reply:
Yes, `StateEventContent` is implemented by `AnyStateEventContent`, which doesn't have a corresponding possibly-redacted type.